ecs-agent/api/ecs/client/ecs_client_factory_test.go

```diff
@@ -0,0 +1,45 @@
+//go:build unit
+// +build unit
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package ecsclient
+
+import (
+	"testing"
+
+	mock_config "github.com/aws/amazon-ecs-agent/ecs-agent/config/mocks"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/ec2"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewECSClientFactory(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	credentialsProvider := credentials.AnonymousCredentials
+	cfgAccessor := mock_config.NewMockAgentConfigAccessor(ctrl)
+	ec2MetadataClient := ec2.NewBlackholeEC2MetadataClient()
+	agentVersion := "0.0.0"
+
+	clientFactory, ok := NewECSClientFactory(credentialsProvider, cfgAccessor, ec2MetadataClient,
+		agentVersion).(*ecsClientFactory)
+	assert.True(t, ok)
+	assert.Equal(t, credentialsProvider, clientFactory.credentialsProvider)
+	assert.Equal(t, cfgAccessor, clientFactory.configAccessor)
+	assert.Equal(t, ec2MetadataClient, clientFactory.ec2MetadataClient)
+	assert.Equal(t, agentVersion, clientFactory.agentVer)
+}
```

Comment:
nit(non-blocking): Perhaps we could also test `NewClient()` as well as `Credentials()` calls just to have better test coverage. Happy for this to be a follow up.

Reply:
Thanks for your comment! I'll add this.

Reply:
Thanks for your comment! I can add this.